### PR TITLE
linux-yocto: Include support for the RealSense camera

### PR DESCRIPTION
**Pull request by telphan**
Closes #127 

General tests:
- [x] Tested image on device

Specific tets:
- [x] RealSense camera is picked shows up as /dev/video[1/0]
- [x] Linux kernel uses the correct driver for the RealSense camera

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Change log
 -----------
 
+* linux-yocto: Include support for the RealSense cmaera [Theodor]
```
Comment (floion) [inline CHANGELOG.md]: typo
> cmaera

Also, the Changelog does not need to have per recipe details, so no need to prefix it with `linux-yocto`

File: layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_camera_formats_linux-yocto_4.4.patch [hunk 2]
```diff
@@ -0,0 +1,169 @@
+From 4b7c35a5363d8d976143896951889dacc57d306b Mon Sep 17 00:00:00 2001
+From: Scott Ware <scott.r.ware@intel.com>
+Date: Mon, 27 Nov 2017 14:19:53 +0000
+Subject: [PATCH 1/4] Add support for RealSense camera formats
+
+Upstream-Status: Backport
```
Comment (floion) [inline layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_camera_formats_linux-yocto_4.4.patch]: needs also signed-ff by you, Theodor, in the correct order.
See https://github.com/resin-os/resin-intel/blob/master/layers/meta-resin-genericx86/recipes-kernel/linux/files/0001-Add-support-for-Quectel-EC20-modem.patch for example

File: layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_hid_linux-yocto_4.4.patch [hunk 3]
```diff
@@ -0,0 +1,181 @@
+From c8f0b4d94aca82dc7f8eae856db98a34941c3ce5 Mon Sep 17 00:00:00 2001
+From: Scott Ware <scott.r.ware@intel.com>
+Date: Mon, 27 Nov 2017 14:34:23 +0000
+Subject: [PATCH 2/4] Realsense HID
+
+Upstream-Status: Backpot
```
Comment (floion) [inline layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_hid_linux-yocto_4.4.patch]: needs also signed-ff by you, Theodor, in the correct order.
See https://github.com/resin-os/resin-intel/blob/master/layers/meta-resin-genericx86/recipes-kernel/linux/files/0001-Add-support-for-Quectel-EC20-modem.patch for example

File: layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_metadata_linux-yocto_4.4.patch [hunk 4]
```diff
@@ -0,0 +1,210 @@
+From c5811c81372e9fe5b5216074a194c1be1ff12d1f Mon Sep 17 00:00:00 2001
+From: Scott Ware <scott.r.ware@intel.com>
+Date: Mon, 27 Nov 2017 15:26:49 +0000
+Subject: [PATCH 3/4] Realsense metadata
+
+Upstream-status: Backport
```
Comment (floion) [inline layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_metadata_linux-yocto_4.4.patch]: needs also signed-ff by you, Theodor, in the correct order.
See https://github.com/resin-os/resin-intel/blob/master/layers/meta-resin-genericx86/recipes-kernel/linux/files/0001-Add-support-for-Quectel-EC20-modem.patch for example

File: layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_powerlinefrequency_control_fix_linux-yocto_4.4.patch [hunk 5]
```diff
@@ -0,0 +1,26 @@
+From e53ff5bf09a54af77f3f2f7c7282e38c33093377 Mon Sep 17 00:00:00 2001
+From: Scott Ware <scott.r.ware@intel.com>
+Date: Mon, 27 Nov 2017 15:27:40 +0000
+Subject: [PATCH 4/4] Fix incomplete PowerLineFrequency enumeration map
+
+Upstream-Status: Backport
```
Comment (floion) [inline layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_powerlinefrequency_control_fix_linux-yocto_4.4.patch]: needs also signed-ff by you, Theodor, in the correct order.
See https://github.com/resin-os/resin-intel/blob/master/layers/meta-resin-genericx86/recipes-kernel/linux/files/0001-Add-support-for-Quectel-EC20-modem.patch for example

File: layers/meta-resin-genericx86/recipes-kernel/linux/linux-yocto_4.4.bbappend [hunk 6]
```diff
@@ -0,0 +1,30 @@
+inherit kernel-resin
```
Comment (floion) [inline layers/meta-resin-genericx86/recipes-kernel/linux/linux-yocto_4.4.bbappend]: there already is a `linux-yocto_%.bbappend` you can use to add these into. Please us that one

File: layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_camera_formats_linux-yocto_4.4.patch (hunk 2 above)
Comment (telphan) [reply layers/meta-resin-genericx86/recipes-kernel/linux/files/realsense_camera_formats_linux-yocto_4.4.patch]: Will add it, but I do not understand why it needs to signed off by me, I did not touch these patches.

File: layers/meta-resin-genericx86/recipes-kernel/linux/linux-yocto_4.4.bbappend (hunk 6 above)
Comment (telphan) [reply layers/meta-resin-genericx86/recipes-kernel/linux/linux-yocto_4.4.bbappend]: This patches are specific to linux-yocto version 4.4

Comment (floion) [reply layers/meta-resin-genericx86/recipes-kernel/linux/linux-yocto_4.4.bbappend]: The kernel version we currently have on Intel NUC is 4.12.12
You suggest we downgrade the kernel to 4.4?

**Comment by telphan** [comment]
Support for the RealSense camera is already present in our kernel.

**Pull request closed without merging**